src/ui/widgets/EmbeddedDisplay/opiParser.ts

```diff
@@ -454,6 +458,146 @@ function opiParseLabelPosition(props: any): string {
   return mapping[num] || "top";
 }
 
+/**
+ * Parses a props object into an Traces object that
+ * contains an array of Trace objects
+ * @param props list of props for this element
+ * @returns a Traces object
+ */
+function opiParseTraces(props: any): Traces {
+  // Find PV value if it's one of the trace properties
+  let pvName = opiParseString(props.pv_name);
+  if (pvName.includes("trace_")) {
+    pvName = opiParseString(props[pvName.slice(2, -1)]);
+  }
+  const count = opiParseNumber(props.trace_count);
+  const traces: Trace[] = [];
+  // Parse all of the 'trace' properties
+  for (let i = 0; i < count; i++) {
+    const trace = parseMultipleNamedProps("trace", props, new Trace(i));
+    traces.push(trace);
+  }
+  return new Traces(count, pvName, traces);
+}
+
+/**
+ * Parses a props object into an AllAxes object that
+ * contains an array of Axis objects
+ * @param props
+ * @returns an Axes object.
+ */
+function opiParseAxes(props: any): Axes {
+  const count = opiParseNumber(props.axis_count);
+  const axes: Axis[] = [];
+  // Parse all of the 'axis' properties
+  for (let i = 0; i < count; i++) {
+    const axis = parseMultipleNamedProps("axis", props, new Axis(i));
+    axes.push(axis);
+  }
+  return new Axes(count, axes);
+}
+
+/**
+ * Converts property name from snake case to camel
+ * case. This is needed to dynamically sort through
+ * all of each traces properties, although it's a
+ * bit messy and clunky.
+ * @param propName string to convert
+ * @returns newly formatted string
+ */
+function snakeCaseToCamelCase(propName: string) {
```

Comment:
It seems we don't really have a 'utils' file for tings like this. Although is the use of this function still specific to parsing? I'm not sure we would need to use it anywhere else. In that case maybe it could be moved to parser.ts?